ceph/rados/core_workflows.py

```diff
@@ -1264,3 +1264,73 @@ def get_osd_stat(self):
         osd_stats = self.run_ceph_command(cmd=cmd)
         log.debug(f" The OSD Statistics are : {osd_stats}")
         return osd_stats
+
+    def run_pool_sanity_check(self):
+        """
+        Runs sanity on the pools after triggering scrub and deep-scrub on pools, waiting 600 Secs
+
+        This method is used to assess the health of Pools after any operation, where in a scrub and deep scrub is
+        triggered, and the method scans the cluster for few health warnings, if generated
+
+        Returns: True/Pass for Pass/Fail
+        """
+        self.run_scrub()
+        self.run_deep_scrub()
+        time.sleep(10)
+
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=600)
+        flag = False
+        while end_time > datetime.datetime.now():
+            status_report = self.run_ceph_command(cmd="ceph report")
+            ceph_health_status = status_report["health"]
+            health_warns = (
+                "PG_AVAILABILITY",
+                "PG_DEGRADED",
+                "PG_RECOVERY_FULL",
+                "TOO_FEW_OSDS",
+                "PG_BACKFILL_FULL",
+                "PG_DAMAGED",
+                "OSD_SCRUB_ERRORS",
+                "OSD_TOO_MANY_REPAIRS",
+                "CACHE_POOL_NEAR_FULL",
+                "SMALLER_PGP_NUM",
+                "MANY_OBJECTS_PER_PG",
+                "OBJECT_MISPLACED",
+                "OBJECT_UNFOUND",
+                "SLOW_OPS",
+            )
+
+            flag = (
+                False
+                if any(
+                    key in health_warns for key in ceph_health_status["checks"].keys()
+                )
+                else True
+            )
+            if flag:
+                log.info("No warnings on the cluster")
+                break
+
+            log.info(
+                f"Observing a health warning on cluster {ceph_health_status['checks'].keys()}"
+            )
+            time.sleep(10)
+
+        if not flag:
+            log.error(
+                "Health warning generated on cluster and not cleared post waiting of 600 seconds"
+            )
+            return False
+
+        log.info("Completed check on the cluster. Pass!")
+        return True
+
+    def get_osd_hosts(self):
```

Comment:
IMHO, later on we should plan to rename this module to `get_osd_hostnames(self)` because just going by the name, one would interpret the osd node objects to be returned instead of hostnames.
Also as we already have this info available with us, we can skip executing a ceph command to reduce execution time, please ignore if irrelevant
```
osd_nodes = ceph_cluster.get_nodes(role="osd")
return [node.hostname for node in osd_nodes]
```

Reply:
On second thought I assume it is being done this way to avoid false failures if a host is actually an OSD host but does not have appropriate labels, so fetching the list from `ceph osd tree` would eliminate the possibility of such a host being missed.

Reply:
Hello Harsh. Great that you noticed. And there's a pretty good reasoning for the same.

The objects that are fetched via : ceph_cluster.get_nodes(role="osd") , The nodes are fetched from the conf file that we provide, not fetching it from the cluster deployed.

In most cases, I bet this would work as expected, without any issues.

But in our scenario, We are not adding all the nodes in the conf file into the cluster. So I need to fetch the host names from the cluster itself. If I call the get_nodes(), it would return me nodes that are not yet added to cluster also, Making verification pretty difficult.